File: 2022/haskell/1_solution.hs
import Data.List
import Data.List.Split

readTotalCalories :: String -> [Int]
readTotalCalories = map (sum . map read) . splitOn [""] . lines


part1 :: String -> Int
part1 = maximum . readTotalCalories

part2 :: String -> Int
part2 = sum . take 3 . reverse . sort . readTotalCalories


main :: IO ()
main = do
    input <- readFile "../input/1_input.txt"
    print $ part1 input -- 70720
    print $ part2 input -- 207148

File: 2022/R/1_solution.R
library(tidyverse)

read_calories <- function(path) {
  read_file(path) %>%
    str_split("\n\n") %>%
    map(str_split, "\n") %>%
    map(~map(.x, as.numeric)) %>%
    map(~map(.x, sum, na.rm = TRUE)) %>% 
    flatten() 
}
  
part_1 <- function(path) {
  read_calories(path) %>%
    reduce(max)
}

part_2 <- function(path) {
  read_calories("../input/1_input.txt") %>%
    as.numeric() %>%
    sort() %>%
    tail(3) %>%
    sum()
}

part_1("../input/1_input.txt") # 70720
part_2("../input/1_input.txt") # 207148
